src/TinyRAM/ExecuteProgram.hs

```diff
@@ -75,13 +77,11 @@ programToMemoryValues params (Program p) =
   if params ^. #wordSize == 0
   then Left "word size must be nonzero but it is zero"
   else
-    case (params ^. #wordSize . #unWordSize) `quotRem` 8 of
-      (bytesPerWord, 0) ->
-        case BS.length p `quotRem` bytesPerWord of
-          (wordsInProgram, 0) ->
-            if wordsInProgram `rem` 2 == 0
-            then Right . MemoryValues . Map.fromList
-                 $ zip [0..] (bytesToWords (params ^. #wordSize) p)
-            else Left "program must consist of an even number of words"
-          _ -> Left "length of program in bytes must be a multiple of the word size in bytes"
+    case (params ^. #wordSize . #unWordSize) `rem` 8 of
+      0 ->
+        Right . MemoryValues . Map.fromList $ zip addresses (bytesToWords (params ^. #wordSize) p)
```

Comment:
I thought we were moving to byte addressed memory? Is that still todo?

Reply:
It's done in following way:
MemoryValues map stores whole words. The keys represent byte addresses of consecutive words (so that key `mod` bytesPerWord == 0). Then load/store instruction just fetches/stores a whole word from the map. In addition to accessing MemoryValues map load.b/store.b instructions have to extract a desired byte from a word.